crate/kmip/src/crypto/mod.rs

```diff
@@ -2,10 +2,13 @@ pub mod cover_crypt;
 pub mod dh_shared_keys;
 pub mod elliptic_curves;
 pub mod generic;
+#[cfg(any(feature = "openssl", feature = "fips"))]
```

Comment:
What if not openssl and not fips ?
There should be a way to derive with Argon2 that does not need fips nor openssl.

Reply:
`derive_key_from_password` that is using `Argon2` in `password_derivation.rs` is using `rand_bytes`, which is a function from openssl.
Maybe it should work without openssl, but today this is not the case

Reply:
ah you are right I forgot about `rand_bytes`
I dont think it should work without it since `rand_bytes` is a secure random generator from fips approved library.